string source_node
string target_node
---
string[] actions
